syncer/server/server.go

```diff
@@ -54,40 +60,55 @@ type Server struct {
 
 	// Wraps the grpc server
 	grpc *grpc.Server
+
+	stopCh chan struct{}
 }
 
 // NewServer new server with Config
 func NewServer(conf *config.Config) *Server {
 	return &Server{
-		conf: conf,
+		conf:   conf,
+		stopCh: make(chan struct{}),
 	}
 }
 
 // Run syncer Server
 func (s *Server) Run(ctx context.Context) {
+	var err error
 	s.initPlugin()
-
-	if err := s.runEtcd(); err != nil {
+	if err = s.initialization(); err != nil {
 		return
 	}
 
-	if err := s.runSerf(); err != nil {
-		return
-	}
+	gopool.Go(syssig.Run)
 
-	if err := s.runGrpc(); err != nil {
+	err = s.startModuleServer(s.agent)
+	if err != nil {
+		s.Stop()
 		return
 	}
 
-	if err := s.createServiceCenter(); err != nil {
+	err = s.startModuleServer(s.etcd)
+	if err != nil {
+		s.Stop()
 		return
 	}
 
-	if err := s.runTicker(); err != nil {
+	err = s.startModuleServer(s.grpc)
+	if err != nil {
+		s.Stop()
 		return
 	}
 
-	s.waitQuit(ctx)
+	s.servicecenter.SetStorage(s.etcd.Storage())
+
+	s.agent.RegisterEventHandler(s)
```

Comment:
This Handler is same with the one in serf/agent.go, but they responsible for different event, I think we should add some comments to them at lease.